internal/pkg/runtime/engine/singularity/process_linux.go

```diff
@@ -567,3 +573,25 @@ func preStartProcess(e *EngineOperations) error {
 
 	return nil
 }
+
+func (e *EngineOperations) getIP() (string, error) {
+	if e.EngineConfig.Network == nil {
+		return "", errors.New("not in net ns")
```

Comment:
I would pass `nil` error to avoid to display warning message
```suggestion
		return "", nil
```